src/containers/Tag/TagList/TagList.tsx

```diff
@@ -44,66 +95,147 @@ export const TagList: React.SFC<TagListProps> = (props) => {
-              <DeleteIcon />
+        <>
+          <Link to={'/tag/' + id + '/edit'}>
+            <IconButton aria-label="Edit" color="default">
+              <EditIcon />
             </IconButton>
-          </TableCell>
-        </TableRow>
+          </Link>
+          <IconButton aria-label="Delete" color="default" onClick={() => deleteHandler(id!)}>
+            <DeleteIcon />
+          </IconButton>
+        </>
       );
+    }
+  }
+
+  function formatTags(tags: Array<any>) {
+    // Should be type tag, but can't import Tag type into file
+    return tags.map((t: any) => {
+      return {
+        label: t.label,
+        description: t.description,
+        operations: getIcons(t.id),
+      };
     });
-  } else {
-    listing = (
-      <TableRow>
-        <TableCell>There are no tags.</TableCell>
-      </TableRow>
-    );
+  }
+
+  const handleSearch = (e: any) => {
+    e.preventDefault();
+    let searchVal = e.target.nameSearch.value.trim();
+    setSearchVal(searchVal);
+  };
+
+  // Making different columns for the table
+  const columns = [
+    {
+      name: 'Name',
+      selector: 'label',
+      sortable: true,
+    },
+    {
+      name: 'Description',
+      selector: 'description',
+      sortable: false,
+    },
+    {
+      name: 'Operations',
+      selector: 'operations',
+      sortable: false,
+    },
+  ];
+
+  // Get tag data and total number of tags.
+  let tagList: any;
+  if (data) {
+    tagList = formatTags(data.tags);
+  }
+
+  let tagCount: number = pageRows;
+  if (countData) {
+    tagCount = countData.countTags;
   }
 
   return (
-    <div>
-      <div className={styles.AddButtton}>
-        <Button variant="contained" color="primary" onClick={() => setNewTag(true)}>
-          New Tag
-        </Button>
+    <>
+      <div className={styles.Header}>
+        <Typography variant="h5" className={styles.Title}>
+          Tags
+        </Typography>
+        <div className={styles.Buttons}>
+          <IconButton className={styles.IconButton} onClick={() => setSearchOpen(!searchOpen)}>
+            <SearchIcon className={styles.SearchIcon}></SearchIcon>
+          </IconButton>
+          <form onSubmit={handleSearch}>
+            <div className={searchOpen ? styles.SearchBar : styles.HideSearchBar}>
+              <InputBase
+                defaultValue={searchVal}
+                className={searchOpen ? styles.ShowSearch : styles.HideSearch}
+                name="nameSearch"
+              />
+              {searchOpen ? (
+                <div className={styles.ResetSearch} onClick={() => setSearchVal('')}>
+                  <Divider orientation="vertical" />
+                  <CloseIcon className={styles.CloseIcon}></CloseIcon>
+                </div>
+              ) : null}
+            </div>
+          </form>
+
+          {/* Not sure how to override Material UI button styling without too much hassle */}
+          <Button
+            style={{
```

Comment:
As the comment says above, putting these styles the CSS file and importing them didn't work properly here (Material UI button overrode my styling). I'll be trying to figure out a solution to this, but in the meantime, the styling is exposed in this file.

Reply:
Let's not add CSS to the component as explained above.